proxy/pkg/config/config.go

```diff
@@ -133,12 +135,46 @@ func New() *Config {
 	return &Config{}
 }
 
+func (c *Config) loadFromFiles() error {
+	paths := os.Getenv("ZDM_CONFIG_FILES")
+	for _, path := range strings.Split(paths, ",") {
+		if len(path) == 0 {
+			continue
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return fmt.Errorf("could not read configuration file %v: %w", path, err)
+		}
+		defer file.Close()
+
+		dec := yaml.NewDecoder(file)
+		if err = dec.Decode(c); err != nil {
+			return fmt.Errorf("could not parse yaml file %v: %w", path, err)
+		}
+	}
+	return nil
+}
+
 // ParseEnvVars fills out the fields of the Config struct according to envconfig rules
 // See: Usage @ https://github.com/kelseyhightower/envconfig
-func (c *Config) ParseEnvVars() (*Config, error) {
+func (c *Config) parseEnvVars() error {
 	err := envconfig.Process("ZDM", c)
 	if err != nil {
-		return nil, fmt.Errorf("could not load environment variables: %w", err)
+		return fmt.Errorf("could not load environment variables: %w", err)
+	}
+
+	return nil
+}
+
+func (c *Config) LoadConfig() (*Config, error) {
+	err := c.parseEnvVars()
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.loadFromFiles()
+	if err != nil {
+		return nil, err
```

Comment:
I don't understand this, what's the goal of parsing both at the same time, shouldn't we parse just the CONFIG_FILES env var first to see if it's set and if it's set we just skip parsing all other env vars and just parse the config files?